### ISSUE TITLE
Dagger does not seem to work with groovy-eclipse-compiler

### ISSUE THREAD

**Issue by kuza55** [google/dagger#122]
Hi,

I have a mixed Java/Groovy project that I was interested in using Groovy in, for which I am using the groovy-eclipse-compiler via the maven compiler plugin.

It seems that this is somehow incompatible; I extracted the simple example from it's parent pom, got it to compile, then attempted to switch compilers and the moment I did so maven stopped generating sources.

Are there any plans to support such use of Dagger 2? Or is there a better way to utilise Dagger 2 in a mixed Java/Groovy project being built using Maven.

 - Alex

**Comment by JakeWharton** [comment]
Groovy does not have annotation processing and as such this will never work.

**Comment by kuza55** [comment]
Thanks for the info; is there a possibility that the Java bits of my application could get to benefit from Dagger?

I'm mostly using Groovy for the business logic bits that have to deal with some highly nested JSON input that is painful to deal with in Java, so not having them participate in Dagger isn't the biggest issue.

I guess I could split it into two separate modules with different compilers.

**Comment by JakeWharton** [comment]
Yes. Splitting it should work fine. As long as the annotated parts which
use Dagger go through a compiler which knows how to run annotation
processors it will work.

**Comment by artem-zinnatullin** [comment]
@kuza55 just keep all `Dagger`'s things: Modules and Components in Java + javac, and inject in Groovy/Java code

**Comment by gk5885** [comment]
I'm trusting @JakeWharton on this and closing this since we can't support this due to a limitation in Groovy.

**Comment by tbroyer** [comment]
Based on https://docs.gradle.org/2.4-rc-1/release-notes#support-for-“annotation-processing”-of-groovy-code I think @JakeWharton is wrong.

**Comment by JakeWharton** [comment]
I'm pretty sure my comment pre-dates 2.4.

**Comment by tbroyer** [comment]
Ha, that would explain it ;-)

**Comment by JakeWharton** [comment]
Yep. By at least 6 to 8 weeks.
